lib/ain-grpc/src/rpc/debug.rs

```diff
@@ -66,4 +82,90 @@ impl MetachainDebugRPCServer for MetachainDebugRPCModule {
 
         Ok(())
     }
+    fn fee_estimate(&self, input: CallRequest) -> RpcResult<FeeEstimate> {
+        let CallRequest {
+            from,
+            to,
+            gas,
+            value,
+            data,
+            access_list,
+            transaction_type,
+            gas_price,
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+            ..
+        } = input;
+
+        let (block_hash, block_number) = self
+            .handler
+            .block
+            .get_latest_block_hash_and_number()
+            .ok_or(Error::Custom(
+                "Error fetching latest block hash and number".to_string(),
+            ))?;
+        let base_fee = self.handler.block.calculate_base_fee(block_hash);
+
+        let TxResponse { used_gas, .. } = self
+            .handler
+            .core
+            .call(EthCallArgs {
+                caller: from,
+                to,
+                value: value.unwrap_or_default(),
+                data: &data.map(|d| d.0).unwrap_or_default(),
+                gas_limit: gas.unwrap_or(MAX_GAS_PER_BLOCK).as_u64(),
```

Comment:
We should throw RPC error if trying to get fee estimate without specifying tx gas.

Reply:
Resolved in [fd6d7c1](https://github.com/DeFiCh/ain/pull/2252/commits/fd6d7c1fc3fb228b20876eba8f0b387e1b9836fb)